addons/web/static/src/core/record_selectors/record_autocomplete.js

```diff
@@ -0,0 +1,140 @@
+/** @odoo-module **/
+
+import { Component } from "@odoo/owl";
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { _t } from "@web/core/l10n/translation";
+import { Domain } from "@web/core/domain";
+import { registry } from "@web/core/registry";
+import { useOwnedDialogs, useService } from "@web/core/utils/hooks";
+
+const SEARCH_LIMIT = 7;
+const SEARCH_MORE_LIMIT = 320;
+
+export class RecordAutocomplete extends Component {
+    static props = {
+        resModel: String,
+        update: Function,
+        multiSelect: Boolean,
+        getIds: Function,
+        value: String,
+        domain: { type: Array, optional: true },
+        context: { type: Object, optional: true },
+        className: { type: String, optional: true },
+        fieldString: { type: String, optional: true },
+    };
+    static components = { AutoComplete };
+    static template = "web.RecordAutocomplete";
+
+    setup() {
+        this.orm = useService("orm");
+        this.nameService = useService("name");
+        this.addDialog = useOwnedDialogs();
+        this.sources = [
+            {
+                placeholder: _t("Loading..."),
+                options: this.loadOptionsSource.bind(this),
+            },
+        ];
+    }
+
+    addNames(nameGets) {
+        const displayNames = {};
+        for (const [id, label] of nameGets) {
+            displayNames[id] = label.split("\n")[0];
+        }
+        this.nameService.addDisplayNames(this.props.resModel, displayNames);
+    }
+
+    getIds() {
+        return this.props.getIds();
```

Comment:
I'm not sure this props makes a lot of sense for a RecordAutocomplete component, as there's already a domain props anyway.